### PR TITLE
WebQuery: rect x and y should indicate the top-left corner of the element

### PR DESCRIPTION
**Pull request by jmoody**
### Motivation

This aligns web query results with UIView results, the behavior of Calabash Android, and with user's expectations.

### TODO

- [x] Fix unit test

### CODE REVIEW

**Comment by jmoody** [comment]
@sapieneptus Jenkins is down.  I have checked UITest to see if it uses the x and y rect coordinates for gestures; it does not as far as I can tell.

**Comment by jmoody** [comment]
My bad; there is a failing unit test.  We have test coverage after all. :)

**Comment by jmoody** [comment]
* Unit test was failing on Xcode 6 only - forcing Travis to use Xcode 7.2
* Jenkins is in a bad state (bad update, bad simulator state, bad code signing env)

**Comment by jmoody** [comment]
Jenkins test this please

**Comment by jmoody** [comment]
Travis is failing because the jobs are not completing fast enough.  Everything is green before the time out.  I am willing to make a note of this and merge.  If we continue to have problems on Travis we can revisit.

**Comment by jmoody** [comment]
@sapieneptus Can you review and merge?

File: calabash/Classes/WebViewQuery/LPWebQuery.m [hunk 1]
```diff
@@ -67,6 +67,8 @@ + (NSArray *) arrayByEvaluatingQuery:(NSString *) query
 
   for (NSDictionary *d in queryResult) {
     NSMutableDictionary *dres = [NSMutableDictionary dictionaryWithDictionary:d];
+
+    // Inexplicably, rect.x and rect.y are the _center_.
```
Comment (sapieneptus) [inline calabash/Classes/WebViewQuery/LPWebQuery.m]: They're calculated to be that way in the server, _non_?

**Comment by sapieneptus** [comment]
@jmoody This seems like a potentially significant change, as I imagine some tests may make use of the `rect` information. I assume this will then be in a `0.2.0` ?

**Comment by jmoody** [comment]
@sapieneptus 

> some tests

You mean user tests?  I checked UITest and Calabash internals; we don't use these values.

I think this is actually a _bug_ in the current implementation.

ATTN @krukow Opinion?

File: calabash/Classes/WebViewQuery/LPWebQuery.m (hunk 1 above)
Comment (jmoody) [inline calabash/Classes/WebViewQuery/LPWebQuery.m]: I think this is an unexpected behavior.  They should be named rect.center_x and rect.center_y.

Comment (jmoody) [inline calabash/Classes/WebViewQuery/LPWebQuery.m]: They are generated by calabash-js.

**Comment by sapieneptus** [comment]
@jmoody It looks like a bug, but what I'm worried about is there may be user tests that rely on it working this way. It looks fine to merge, but I would recommend this constituting a 'major' release (0.2.0) as it is altering (technically fixing) a very core part of the server upon which users may be relying.

**Comment by jmoody** [comment]
Ugh.  I am not ready to release 0.20.0.  The next release is a patch release.  Let's discuss when you arrive in Aarhus.  I don't disagree with you - it _might_ break backward compatibility.

**Comment by jmoody** [comment]
@sapieneptus I discussed with @jescriba, @acroos, and @Lagoja today.

We think at 0.19.1 release is OK because it fixes a bug in our published API - the API itself did not change.

**Comment by jmoody** [comment]
It will, of course, be noted in the release notes.

**Comment by sapieneptus** [comment]
Remember that awful period of time where browsers actually tried to render HTML correctly and enforce the rules? Sites had already come to rely on improper HTML and broke as a result.

In the same spirit, I respectfully disagree in principle that it doesn't warrant a full 0.2.0 release. In practice, however, I have no idea if anyone actually is relying on those values. It might be totally fine. If it were known that this would cause mass breakage I'd argue more strongly, but in this case I'm content to just let it be known that I think it violates the spirit and principles of semantic versioning.

**Pull request merged**